mh/database.py

```diff
@@ -0,0 +1,328 @@
+        for i in range(1, 3)
+    ])
+    servers.extend([
+        Server("Greed{}".format(i), ServerType.RECRUITING)
+        for i in range(1, 5)
+    ])
+    return servers
+
+
+class TempDatabase(object):
+    """A temporary database.
+
+    TODO:
+     - Finish the implementation
+     - Make this thread-safe
+    """
+
+    def __init__(self):
+        self.consoles = {
+            # Online support code => Capcom IDs
+        }
+        self.sessions = {
+            # PAT Ticket => Owner's session
+        }
+        self.capcom_ids = {
+            # Capcom ID => Owner's session
+            "C9I7D4": {"name": "Cid", "session": None},
+            "D9R7K4": {"name": "Drakea", "session": None},
+        }
+        self.servers = new_servers()
+
+    def get_support_code(self, session):
+        """Get the online support code or create one."""
+        support_code = session.online_support_code
+        if support_code is None:
+            while True:
+                support_code = new_random_str(11)
+                if support_code not in self.consoles:
+                    session.online_support_code = support_code
+                    break
+
+        # Create some default users
+        if support_code not in self.consoles:
+            self.consoles[support_code] = [
+                "C9I7D4", "D9R7K4", "******",
+                "******", "******", "******"
+            ]
+        return support_code
+
+    def new_pat_ticket(self, session):
+        """Generates a new PAT ticket for the session."""
+        while True:
+            session.pat_ticket = new_random_str(11)
+            if session.pat_ticket not in self.sessions:
+                break
+        self.sessions[session.pat_ticket] = session
+        return session.pat_ticket
+
+    def use_user(self, session, index, name):
+        """Use Capcom ID from the slot or create one if empty"""
+        assert 1 <= index <= 6, "Invalid Capcom ID slot"
+        index -= 1
+        users = self.consoles[session.online_support_code]
+        while users[index] == "******":
+            capcom_id = new_random_str(6)
+            if capcom_id not in self.capcom_ids:
+                users[index] = capcom_id
+                break
+        else:
+            capcom_id = users[index]
+            not_in_use = self.capcom_ids[capcom_id]["session"] is None
+            assert not_in_use, "Capcom ID is already in use"
+        name = name or self.capcom_ids[capcom_id]["name"]
+        self.capcom_ids[capcom_id] = {"name": name, "session": session}
+        session.capcom_id = capcom_id
+        session.hunter_name = name
+
+    def get_session(self, session):
+        """Returns existing PAT session or the current one."""
+        return self.sessions.get(session.pat_ticket, session)
+
+    def del_session(self, session):
+        """Delete the session from the database."""
+        # TODO: Find a good place to purge old tickets
```

Comment:
About this comment, I think we are going to need some kind of scheduler or something a long those line to deal with this kind of task. Also, we are going to need it in order to simulate the day/night cycle with ticks

Reply:
I do agree, we'll have to tackle this issue, at some point.